### PR TITLE
Operator field path evaluation

### PR DESCRIPTION
**Pull request by wcekan**
Required for FilterExpressionChecks that process predicates with relationships like
`this.author.book`

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by DeathByTape** [comment]
👍

File: elide-datastore/elide-datastore-hibernate3/src/main/java/com/yahoo/elide/datastores/hibernate3/filter/CriterionFilterOperation.java [hunk 1]
```diff
@@ -117,6 +117,10 @@ public Criterion apply(Predicate predicate) {
                 return Restrictions.gt(alias, predicate.getValues().get(0));
             case GE:
                 return Restrictions.ge(alias, predicate.getValues().get(0));
+            case TRUE:
```
Comment (aklish) [inline elide-datastore/elide-datastore-hibernate3/src/main/java/com/yahoo/elide/datastores/hibernate3/filter/CriterionFilterOperation.java]: Should also cover the IN and NOT cases like for HQL.

Comment (wcekan) [inline elide-datastore/elide-datastore-hibernate3/src/main/java/com/yahoo/elide/datastores/hibernate3/filter/CriterionFilterOperation.java]: Done.

**Comment by DeathByTape** [comment]
👍

**Comment by aklish** [comment]
👍 But we still need to investigate why we need '.' separated paths instead of using the Predicate path.  Something isn't right.

**Comment by aklish** [comment]
👍

**Comment by DeathByTape** [comment]
👍

**Comment by aklish** [comment]
👍

**Pull request merged**